flax/nnx/nnx/nn/normalization.py

```diff
@@ -633,3 +631,201 @@ def __call__(self, x, mask: tp.Optional[jax.Array] = None):
       self.dtype,
       self.epsilon,
     )
+
+class GroupNorm(Module):
+  """Group normalization (arxiv.org/abs/1803.08494).
+
+  This op is similar to batch normalization, but statistics are shared across
+  equally-sized groups of channels and not shared across batch dimension.
+  Thus, group normalization does not depend on the batch composition and does
+  not require maintaining internal state for storing statistics.
+  The user should either specify the total number of channel groups or the
+  number of channels per group.
+
+  .. note::
+    LayerNorm is a special case of GroupNorm where ``num_groups=1``.
+
+  Example usage::
+
+    >>> from flax import nnx
+    >>> import jax
+    >>> import numpy as np
+
+    >>> x = jax.random.normal(jax.random.key(0), (3, 4, 5, 6))
+    >>> layer = nnx.GroupNorm(num_features=6, num_groups=3, rngs=nnx.Rngs(0))
+    >>> nnx.state(layer)
+    State({
+      'bias': VariableState(
+        type=Param,
+        value=Array([0., 0., 0., 0., 0., 0.], dtype=float32)
+      ),
+      'scale': VariableState(
+        type=Param,
+        value=Array([1., 1., 1., 1., 1., 1.], dtype=float32)
+      )
+    })
+    >>> y = layer(x)
+
+    >>> y = nnx.GroupNorm(num_features=6, num_groups=1, rngs=nnx.Rngs(0))(x)
+    >>> y2 = nnx.LayerNorm(num_features=6, reduction_axes=(1, 2, 3), rngs=nnx.Rngs(0))(x)
+    >>> np.testing.assert_allclose(y, y2)
```

Comment:
Use `...` to express empty lines:
```suggestion
    >>> from flax import nnx
    >>> import jax
    >>> import numpy as np
    ...
    >>> x = jax.random.normal(jax.random.key(0), (3, 4, 5, 6))
    >>> layer = nnx.GroupNorm(num_features=6, num_groups=3, rngs=nnx.Rngs(0))
    >>> nnx.state(layer)
    State({
      'bias': VariableState(
        type=Param,
        value=Array([0., 0., 0., 0., 0., 0.], dtype=float32)
      ),
      'scale': VariableState(
        type=Param,
        value=Array([1., 1., 1., 1., 1., 1.], dtype=float32)
      )
    })
    >>> y = layer(x)
    ...
    >>> y = nnx.GroupNorm(num_features=6, num_groups=1, rngs=nnx.Rngs(0))(x)
    >>> y2 = nnx.LayerNorm(num_features=6, reduction_axes=(1, 2, 3), rngs=nnx.Rngs(0))(x)
    >>> np.testing.assert_allclose(y, y2)
```